_posts/2020-10-01-intro-to-hive-connector.md

```diff
@@ -0,0 +1,101 @@
+a long time because the data that ETL would operate on was big. The primary goal
+in early iterations of these systems was simply throughput over large volumes of
+data while maintaining fault-tolerance. Over time, more businesses wanted to
+run fast interactive queries over their Big Data sets instead of running jobs
+that took hours and produced possibly undesirable results. Upon Presto's
+creation, 300 petabytes of data and all the metadata for this data that existed. 
+The data in storage or in metastore didn't need to be fixed, and only the
+runtime that executed Hive queries needed replacement. So the creators took this
+into account and built the Presto engine to utilize the existing metastore
+metadata and files residing in storage, and the presto runtime replaced the Hive
+runtime responsible for analyzing the data.
+
+![](/assets/blog/intro-to-hive-connector/presto.png)
+
+Notice that the only change in the Presto architecture is the runtime, and
+the Hive metastore still exists along with the storage. This was not by
+accident. This design was to address a common pattern that many companies faced,
+and it simplified the migration from using Hive to using Presto. Since the 
+central model used to store the metadata makes use of the Hive metastore
+regardless of the storage component used, it seemed fitting that the connector 
+received the name hive-connector since the metastore can be configured over
+many different storage solutions. 
+
+Where the confusion tends to come from, is many people search for a connector
+from the context of the storage systems they want to query and may not even be 
+aware that the metastore is a necessity or even exists. A new person to Presto
+will typically look for something like an s3-connector or gcs-storage-connector
+when all they really need is the hive-connector which connects them to a Hive 
+metastore service, configured to save metadata over the filesystem they are
+writing to. 
+
+To be clear, the Hive metastore service is the only Hive binary used
+in the entire Presto ecosystem. The Hive metastore is actually a simple service
+with a binary api called Thrift. This service makes updates to the metadata
+state which gets stored into a RDBMS such as Postgres, Mysql, or Mariadb. There
+are other variations of the Hive metastore such as AWS Glue which acts as a drop
+in substitution.
+
+To drive this point home, I have created a Katacoda exercise. In the following
+scenario, I have created a docker environment that creates four docker
+container nodes based on the following images: `presto`, `minio`, `hive
+-metastore`, and `mariadb`. Presto will be _the runtime_ in this scenario that
+replaces Hive. Minio is an open-source S3 compatible object storage system that
+acts as _the storage_ component. Finally, we will run a `hive-metastore`
+instance that connects to a `maraidb` instance. You will be able to play around
```

Comment:
> maraidb

typo

Reply:
MariaDB

Reply:
This particular instance is referencing the docker image name